### ISSUE TITLE
Builder does not stand in the right place to build

### ISSUE THREAD

**Issue by PenguinMan98** [ldtteam/minecolonies#830]
### Minecolonies version

version 1.11-0.6.3360

### Expected behavior

- Builder determines where the new construction is by where the builders hut is in relation to it, then moves there to build.

### Actual behaviour

- Builder seems to be taking reference from where the town hall is to the new construction but then going to that location starting at the builders hut.  As a result, he's always off from where he needs to be by the distance of the hut to the town hall

### Steps to reproduce the problem

1. Mark the town hall location. Make it top priority.
2. Mark the Builders Hut location.
3. Give the builder the stuff required to build the town hall first.

**Comment by SulpherStaer** [comment]
I'm not sure what exactly you mean, but the builder indeed stands quite far away from where he's actually building... which is annoying.

**Comment by Raycoms** [comment]
What you mean to say is that the builder walks first to the townhall and
then to the construction site?
Or first to the chest of the townhall and then to the position he
constructs the contruction site at?

1) On purpose if the townhall is kinda on the way.
2) on purpose to avoid himself closing himself in.

2017-04-18 21:13 GMT-03:00 Silvester <notifications@github.com>:

**Comment by SulpherStaer** [comment]
Avoid closing himself in.. by building 30 blocks away?... It's almost 2 chunks sometimes. I can never find my builder in a forested area when he's building.

I think the tolerance is a bit to high. I mean, we know exactly what is being built, we know the size. There must be a way to lower the tolerance on how far away the builder is compared to the build.

**Comment by Raycoms** [comment]
he we make it the easier way atm. Longest side from the center of the
building.
As soon as I finished my thesis I will improve this as well a bit. I have
some nice ideas.

2017-04-18 21:23 GMT-03:00 Silvester <notifications@github.com>:

**Comment by PenguinMan98** [comment]
What I mean is, let's assume a simple flat grid, top-down. I place my town hall at 0,0.  I place my builders hut at 2,2.  I want to build my farm at 2,-2.  The distance vector to the new construction is calculated from the position of the town hall, down two and right two.  But the builder departs using that vector from the builders hut, travelling down two and right two to arrive at grid 4,0 where he will proceed to build the building.  Does that make sense?

**Comment by PenguinMan98** [comment]
If I wanted to build the building at -2, 2, then the builder would build it standing at 0,4 and if I tried to build it at -2,-2, then my theory is he will try to build it while standing inside the town hall!

**Comment by SulpherStaer** [comment]
I have no idea what you mean.

**Comment by PenguinMan98** [comment]
Let me try again.

Let's say I built my townhall in the game at the very center of the map.  x=0 z=0. 
I then build my builders hut 30 blocks away at x=30 z=0.  
My theory is, wherever the building is the builder is supposed to build, he will always be standing 30 blocks to the right of it because the builders hut is 30 blocks away from the town hall.

If I put the builders hut 40 blocks away from the town hall at x=-40 z=0, then my theory is that the builder will always be standing 40 blocks to the left of where he should be building because the builders hut is 40 blocks away from the town hall.

**Comment by Raycoms** [comment]
That's a coincidence. He always goes to the offset of the max(length,
width) of the hut block of the building, into a certain direction (but he
tries all 4 to find a good location)

2017-04-19 16:28 GMT-03:00 Joe <notifications@github.com>:

**Comment by PenguinMan98** [comment]
Maybe so. What I'm saying is that he's starting from the wrong reference point (builders hut instead of the town hall), and so, he never goes to the right spot, disappearing into the woods outside the settlement to build, or out in the ocean.  Or in my case, skirting the edge of a ravine.

**Comment by neadlak** [comment]
Wouldn't this be simple to test?

Get 2 builder huts online with different placements (different offsets from town hall as you claim)
Start 2 build jobs of identical buildings.
See where the builders go relative to their jobs.
Turn on creative and watch them from above; take a screenshot to show their distance-from-center.

What you should see is that they stand approximately the same distance from center of their target, regardless of how far their hut is from town hall. This assumes they have adequate room around their build target from which to work.

If working as intended: Any offset phenomenon you think you see is actually intended behavior so that builders are always working from outside the possible space that they might place blocks (and suffocate themselves).

If we have a bug: one builder will be obviously further from the build target.

**Comment by PenguinMan98** [comment]
Well, my theory seems bunk but the builder still doesn't seem to pick a spot to stand very intelligently.  It seems they always choose a spot on the far side of the construction project from the town hall which means they almost always path through the build site before stopping on the far side to begin work.  Also, the further the construction is from the town hall, the further they seem to go past the site before stopping to build.

**Comment by PenguinMan98** [comment]
Another thing I've noticed is that they will prefer the far side from the town hall even if that side is the least hospitable for them to stand.  My builders hut is on the seashore. It has land on three sides and ocean on the fourth. When upgrading it, the builder always dives into the ocean and builds from there.

**Comment by Raycoms** [comment]
closed in favor of #943
